app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -590,14 +591,14 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Activit
 
         searchFeature.set(
             feature = SearchFeature(store, customTabSessionId) { request, tabId ->
-                val parentSession = sessionManager.findSessionById(tabId)
+                val parentSession = store.state.findTabOrCustomTab(tabId)
                 val useCase = if (request.isPrivate) {
                     requireComponents.useCases.searchUseCases.newPrivateTabSearch
                 } else {
                     requireComponents.useCases.searchUseCases.newTabSearch
                 }
 
-                if (parentSession?.isCustomTabSession() == true) {
+                if (parentSession is CustomTabSessionState) {
```

Comment:
This part is strange. Why did we do this? 🤔  How can the parent be a custom tab?

Reply:
Ah I see this is just naming, parent here is just what triggered the search so this should be fine.